packages/craco/lib/features/webpack/eslint.js

```diff
@@ -48,80 +45,100 @@ function extendsEslintConfig(loader, eslintConfig, context) {
     } else {
-        loader.options = {
+        webpackEslintConfig.options = {
             ignore: true
         };
     }
 
     log("Overrided ESLint config to enable an ignore file.");
 }
 
-function applyLoaderOptions(loader, loaderOptions, context) {
-    if (isFunction(loaderOptions)) {
-        loader.options = loaderOptions(loader.options || {}, context);
+function applyOptionsConfiguration(webpackEslintConfig, pluginOptions, context) {
+    if (isFunction(pluginOptions)) {
+        webpackEslintConfig.options = pluginOptions(webpackEslintConfig.options || {}, context);
 
-        if (!loader.options) {
-            throw new Error("craco: 'eslint.loaderOptions' function didn't return a loader config object.");
+        if (!webpackEslintConfig.options) {
+            throw new Error("craco: 'eslint.pluginOptions' function didn't return a config object.");
         }
     } else {
         // TODO: ensure is otherwise a plain object, if not, log an error.
-        loader.options = deepMergeWithArray(loader.options || {}, loaderOptions);
+        webpackEslintConfig.options = deepMergeWithArray(webpackEslintConfig.options || {}, pluginOptions);
     }
 
-    log("Applied ESLint loader options.");
+    log("Applied ESLint plugin options.");
 }
 
 function overrideEsLint(cracoConfig, webpackConfig, context) {
     if (cracoConfig.eslint) {
-        const { isFound, match } = getLoader(webpackConfig, loaderByName("eslint-loader"));
+        const { isFound, webpackEslintConfig, disableEsLint } = loadWebpackEslintPlugin(cracoConfig, webpackConfig);
 
         if (!isFound) {
-            logError("Cannot find ESLint loader (eslint-loader).");
-
             return webpackConfig;
         }
 
-        const { enable, mode, loaderOptions } = cracoConfig.eslint;
+        const { enable, mode, pluginOptions } = cracoConfig.eslint;
 
         if (enable === false) {
-            disableEslint(webpackConfig);
+            const { hasRemovedAny } = disableEsLint();
+
+            if (hasRemovedAny) {
+                log("Disabled ESLint.");
+            } else {
+                logError("Couldn't disabled ESLint.");
+            }
 
             return webpackConfig;
         }
 
-        enableEslintIgnoreFile(match.loader);
+        enableEslintIgnoreFile(webpackEslintConfig);
 
         if (mode === ESLINT_MODES.file) {
-            useEslintConfigFile(match.loader);
+            useEslintConfigFile(webpackEslintConfig);
         } else {
-            extendsEslintConfig(match.loader, cracoConfig.eslint, context);
+            extendsEslintConfig(webpackEslintConfig, cracoConfig.eslint, context);
         }
 
-        if (loaderOptions) {
-            applyLoaderOptions(match.loader, loaderOptions);
+        if (pluginOptions) {
+            applyOptionsConfiguration(webpackEslintConfig, pluginOptions);
         }
     }
 
     return webpackConfig;
 }
 
+function loadWebpackEslintPlugin(cracoConfig, webpackConfig) {
```

Comment:
Since this feature doesn't support 2 strategies anymore (loader and plugin) this code could be moved back to the function that actually implement the features like it was done for CRA 3.*